external-crates/move/move-execution/v1/crates/move-bytecode-verifier/src/absint.rs

```diff
@@ -3,18 +3,37 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use move_binary_format::{
-    binary_views::FunctionView,
-    control_flow_graph::{BlockId, ControlFlowGraph},
+    control_flow_graph::{BlockId, ControlFlowGraph, VMControlFlowGraph},
     errors::PartialVMResult,
-    file_format::{Bytecode, CodeOffset},
+    file_format::{
+        AbilitySet, Bytecode, CodeOffset, CodeUnit, FunctionDefinitionIndex, FunctionHandle,
+        Signature,
+    },
+    CompiledModule,
 };
 use move_bytecode_verifier_meter::{Meter, Scope};
 use std::collections::BTreeMap;
 
+/// A `FunctionContext` holds all the information needed by the verifier for `FunctionDefinition`.`
+/// A control flow graph is built for a function when the `FunctionContext` is created.
+pub struct FunctionContext<'a> {
```

Comment:
The fact that we are replacing a type that we previously part of `move_binary_format` (not subject to versioning) with one that is is interesting -- could we have put this type in the binary format to avoid having to duplicate it?

Reply:
We could. But I think conceptually it should be duplicated. It is only used by the verifier and could potentially contain verifier only changes (which are subject to versioning)